microsoft_teams/icon_microsoft_teams/util/azure_ad_utils.py

```diff
@@ -314,12 +316,77 @@ def enable_teams_for_group(logger, connection, group_id):
         sleep(10)
         result = requests.put(endpoint, json=payload, headers=headers)
         if result.status_code == 201:
-            logger.info(f"Team was enabled successfully")
+            logger.info("Team was enabled successfully")
             return True
 
     raise PluginException(cause=f"Could not enable Teams for group with ID: {group_id}.",
                           assistance="This may be due to a replication delay in Azure. Please try again later, or "
-                                     "enable the team manually. If this problem persists, contact Rapid7 support with the "
-                                     "following information:",
+                                     "enable the team manually. If this problem persists, contact Rapid7 support with "
+                                     "the following information:",
                           data=f"Status Code: {result.status_code}\n"
                                f"Response:\n{result.text}")
+
+
+def add_user_to_owners(logger: Logger, connection: komand.connection, group_id: str, user_id: str) -> bool:
+    endpoint = f"https://graph.microsoft.com/beta/groups/{group_id}/owners/$ref"
+    logger.info(f"Adding user to group owners with: {endpoint}.")
+
+    result = requests.post(
+        endpoint,
+        json={
+            "@odata.id": f"https://graph.microsoft.com/beta/users/{user_id}"
+        },
+        headers=connection.get_headers()
+    )
+    try:
+        result.raise_for_status()
+    except Exception as e:
+        raise PluginException(
+            cause=f"Unable to add user to group owners:\nUser:{user_id}\nGroup:{group_id}.",
+            assistance=f"{result.text}",
+            data=e
+        )
+
+    if result.status_code == 204:
+        logger.info("User was added successfully.")
+        return True
+
+    raise PluginException(
+        cause=f"Unexpected response from server when adding user to group owners. Response code: {result.status_code}.",
+        assistance="Please contact Rapid7 support with the following error information.",
+        data=result.text
+    )
+
+
+def add_user_to_channel(logger: Logger, connection: komand.connection, group_id: str, channel_id: str,
+                        user_id: str) -> bool:
+    endpoint = f"https://graph.microsoft.com/beta/teams/{group_id}/channels/{channel_id}/members/"
+    logger.info(f"Adding user to channel with: {endpoint}.")
+
+    result = requests.post(
+        endpoint,
+        json={
+            "@odata.type": "#microsoft.graph.aadUserConversationMember",
+            "roles": [],
+            "user@odata.bind": f"https://graph.microsoft.com/beta/users/{user_id}"
+        },
+        headers=connection.get_headers()
+    )
+    try:
+        result.raise_for_status()
+    except Exception as e:
+        raise PluginException(
+            cause=f"Unable to add user to channel:\nUser:{user_id}\nChannel:{channel_id}.",
```

Comment:
Is there an error response for a user that is already added? if so, we can check that and still succeed if the user is already in the channel. And fail for other unknowns